File: 2010_2011_mf_sujet_4_convergences_de_processus/plim.py
import numpy as np


def plim(p, q):
    """Stationary distribution of the M/Geo/1 queue (truncated to 15 states)."""
    res = np.zeros(15)
    res[0] = 1 - p / q
    idx = np.arange(1, 15)
    res[1:] = (1 / (1 - q)) * (1 - p / q) * (p / q)**idx * ((1 - q) / (1 - p))**idx
    return res

File: 2010_2011_mf_sujet_4_convergences_de_processus/plim.m
function [res] = plim(p,q )
%UNTITLED5 Summary of this function goes here
%   Detailed explanation goes here
res=zeros(1,15);
res(1)=1-p/q;
res([2:15])=1/(1-q)*(1-p/q)*(p/q).^[1:14].*((1-q)/(1-p)).^[1:14];
end

